src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -1661,9 +1661,7 @@ private void getCallInfo(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_RESO
             pResult->methodFlags = FilterNamedIntrinsicMethodAttribs(pResult->methodFlags, methodToCall);
 
             var targetDetails = _compilation.TypeSystemContext.Target;
-            if (targetDetails.Architecture == TargetArchitecture.X86
-                && targetDetails.OperatingSystem == TargetOS.Windows
-                && targetMethod.IsNativeCallable)
+            if (targetDetails.Architecture == TargetArchitecture.X86 && targetMethod.IsNativeCallable)
```

Comment:
>if [](start = 12, length = 2)

Should also update [https://github.com/dotnet/runtime/blob/8772dd31552d110aa217463c064f06b81c2a0e7d/src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs#L2921-L2932](https://github.com/dotnet/runtime/blob/8772dd31552d110aa217463c064f06b81c2a0e7d/src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs#L2921-L2932)

Reply:
Fixed